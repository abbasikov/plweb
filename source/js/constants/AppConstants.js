/*
 * AppConstants
 * These are the variables that determine what our central data store (reducer.js)
 * changes in our state. When you add a new action, you have to add a new constant here
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'YOUR_ACTION_CONSTANT';
 */

// The initial application state
export const INITIAL_STATE = {
    formState: {
        username: '',
        password: ''
    },
    sendingLoginRequest: false,
    loggedIn: false,
    errorMessage: '',
    currentUser:{},
    loginError:'',
    homeScreenLoading:false,
    feeds:[]
};

export const SERVER_URL = 'https://demo5100262.mockable.io';
export const CHANGE_FORM = 'CHANGE_FORM';
export const SET_AUTH = 'SET_AUTH';
export const SET_FEEDS = 'SET_FEEDS';
export const SENDING_REQUEST = 'SENDING_REQUEST';
export const SENDING_LOGIN_REQUEST = 'SENDING_LOGIN_REQUEST';
export const SET_ERROR_MESSAGE = 'SET_ERROR_MESSAGE';
export const FETCH_POSTS = 'FETCH_POSTS';
export const CREATE_POST = 'CREATE_POST';
export const FETCH_POST = 'FETCH_POST';
export const DELETE_POST = 'DELETE_POST';
export const SELECT_BOOK = "SELECT_BOOK";
export const LOGIN_USER = "LOGIN_USER";
export const SET_HOME_SCREEN_LOADING = "SET_HOME_SCREEN_LOADING";