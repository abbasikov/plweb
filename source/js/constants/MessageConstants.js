/*
 * MessageConstants
 * These are the variables that contain the text which is displayed on certain errors
 *
 * Follow this format:
 * export const YOUR_CONSTANT = 'Your message text';
 */
export const FIELD_MISSING = 'Please fill out the entire form.';
export const WRONG_PASSWORD = 'Wrong password.';
export const USER_NOT_FOUND = 'This username does not exist.';
export const USERNAME_TAKEN = 'Sorry, but this username is already taken';
export const GENERAL_ERROR = 'Something went wrong, please try again';
export const LOADING_TEXT= "Please wait...";