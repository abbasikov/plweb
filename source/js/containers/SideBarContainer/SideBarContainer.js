import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

class SideBarContainer extends React.Component {

    constructor(props){
        super(props);
    }


    componentWillReceiveProps(nextProps){

    }


    render() {

        return (
            <div className="sidebar sidebar-opposite sidebar-default sidebar-separate">
                <div className="sidebar-content">

                    <div className="sidebar-category">
                        <div className="category-content">
                            <form action="#">
                                <div className="has-feedback has-feedback-left">
                                    <input type="search" className="form-control" placeholder="Search"/>
                                    <div className="form-control-feedback">
                                        <i className="icon-search4 text-size-base text-muted"></i>
                                    </div>
                                </div>
                            </form>
                        </div>
                    </div>

                </div>
            </div>
        );
    }
}

function mapStateToProps(state) {
    //Whatever is returned from this function will end up
    // as props inside of this container component
    return {

    }
}

function mapDispatchToProps(dispatch) {
    //return bindActionCreators({setAuthState}, dispatch);
}


export default connect(mapStateToProps, null)(SideBarContainer);