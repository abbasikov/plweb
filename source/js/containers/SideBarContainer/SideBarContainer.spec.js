import React from 'react';
import {shallow, render, mount } from 'enzyme';
import toJson  from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { INITIAL_STATE } from '../../constants/AppConstants';
import SideBarContainer from './SideBarContainer';

const mockStore = configureMockStore([ thunk ]);

describe('SideBarContainer', function() {
    var store;
    var container;

    beforeEach(function() {
        store = mockStore(INITIAL_STATE);
    });

    describe('#render', () => {
        it('initial state', function() {
            container = render(<Provider store={store}><SideBarContainer /></Provider>);
            expect(toJson(container)).toMatchSnapshot();
        });
    });

});