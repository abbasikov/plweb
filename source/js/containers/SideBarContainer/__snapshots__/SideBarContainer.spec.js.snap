// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBarContainer #render initial state 1`] = `
<div
  class="sidebar sidebar-opposite sidebar-default sidebar-separate"
>
  <div
    class="sidebar-content"
  >
    <div
      class="sidebar-category"
    >
      <div
        class="category-content"
      >
        <form
          action="#"
        >
          <div
            class="has-feedback has-feedback-left"
          >
            <input
              class="form-control"
              placeholder="Search"
              type="search"
            />
            <div
              class="form-control-feedback"
            >
              <i
                class="icon-search4 text-size-base text-muted"
              />
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
